test/spec/modules/taboolaBidAdapter_spec.js

```diff
@@ -0,0 +1,450 @@
+        'bcat': [],
+        'badv': [],
+        'user': {
+          'buyeruid': 0,
+          'ext': {},
+        },
+        'regs': {'coppa': 0, 'ext': {}}
+      };
+
+      const res = spec.buildRequests([defaultBidRequest], commonBidderRequest);
+
+      expect(res.url).to.equal(`${END_POINT_URL}/${commonBidRequest.params.publisherId}`);
+      expect(res.data).to.deep.equal(JSON.stringify(expectedData));
+    })
+
+    it('should pass optional parameters in request', function () {
+      const optionalParams = {
+        badv: ['adadadbcd.com'],
+        bcat: ['IAB25', 'IAB7-39'],
+        bidfloor: 0.25,
+        bidfloorcur: 'EUR'
+      }
+      const bidRequest = {
+        ...defaultBidRequest,
+        params: {...commonBidRequest.params, ...optionalParams}
+      }
+
+      const res = spec.buildRequests([bidRequest], commonBidderRequest)
+      const resData = JSON.parse(res.data)
+      expect(resData.bcat).to.deep.equal(['IAB25', 'IAB7-39'])
+      expect(resData.badv).to.deep.equal(['adadadbcd.com'])
+      expect(resData.imp[0].bidfloor).to.deep.equal(0.25)
+      expect(resData.imp[0].bidfloorcur).to.deep.equal('EUR')
+    });
+
+    it('should pass bidder timeout', function () {
+      const bidderRequest = {
+        ...commonBidderRequest,
+        timeout: 500
+      }
+      const res = spec.buildRequests([defaultBidRequest], bidderRequest)
+      const resData = JSON.parse(res.data)
+      expect(resData.tmax).to.equal(500)
+    });
+
+    describe('handle privacy segments when building request', function () {
+      it('should pass GDPR consent', function () {
+        const bidderRequest = {
+          refererInfo: {
+            referer: 'https://example.com/'
+          },
+          gdprConsent: {
+            gdprApplies: true,
+            consentString: 'consentString',
+          }
+        };
+
+        const res = spec.buildRequests([defaultBidRequest], bidderRequest)
+        const resData = JSON.parse(res.data)
+        expect(resData.user.ext.consent).to.equal('consentString')
+        expect(resData.regs.ext.gdpr).to.equal(1)
+      });
+
+      it('should pass us privacy consent', function () {
+        const bidderRequest = {
+          refererInfo: {
+            referer: 'https://example.com/'
+          },
+          uspConsent: 'consentString'
+        }
+        const res = spec.buildRequests([defaultBidRequest], bidderRequest)
+        const resData = JSON.parse(res.data)
+        expect(resData.regs.ext.us_privacy).to.equal('consentString')
+      });
+
+      it('should pass coppa consent', function () {
+        config.setConfig({coppa: true})
+
+        const res = spec.buildRequests([defaultBidRequest], commonBidderRequest)
+        const resData = JSON.parse(res.data);
+        expect(resData.regs.coppa).to.equal(1)
+
+        config.resetConfig()
+      });
+    })
+
+    describe('handle userid ', function () {
+      it('should get user id from local storage', function () {
```

Comment:
please make sure your tests work post https://github.com/prebid/Prebid.js/pull/8279

Reply:
seems like other adapters tests are failing in my last 2 builds 
there is away to rerun the job without making a new push to my brunch?
BTW it passed now...

Reply:
Hi [patmmccann](https://github.com/patmmccann) - did you had a chance to look at it after I made the change?